spec.bs

```diff
@@ -1011,25 +1025,13 @@ An <dfn>navigation API method navigation</dfn> is a [=struct=] with the followin
 
        <p class="note">This can occur if a previously-queued-up traversal already took us to this session history entry. In that case that previous traversal will have dealt with |ongoingNavigation| already.
 
-    1. Let |targetStep| be null.
+    1. [=Apply the history step=] given by |targetEntry|'s [=session history entry/step=] to |traversable|, with <i>[=apply the history step/checkForUserCancellation=]</i> set to true, <i>[=apply the history step/unsafeNavigationStartTime=]</i> set to |unsafeNavigationStartTime|, <i>[=apply the history step/sourceSnapshotParams=]</i> set to |sourceSnapshotParams|, and <i>[=apply the history step/initiatorToCheck=]</i> set to |navigable|.
```

Comment:
Before this, should there be a check that the target entry is within the same contiguous same-origin block as the source? The intent of the API is to limit navigations & visibility to those entries right? It seems possible that another navigation may have changed those conditions

Reply:
Interesting. The only way that could happen would be via the replace-splitting case, I think. But yeah, it seems bad, and deterministically triggerable. Let me file a new issue.

Reply:
Hmm, I tried writing it up and I can't actually figure out how to trigger it. To do the replace navigation you have to do an actual full cross-document navigation. But then when you go back to a page which intends to do `traverseTo(nowInvalidKey)`, that page will get the updated `entries()` list, so the early checks inside `traverseTo()` should catch it.

In other words I can't figure out how to split the history during the period between `traverseTo()` being called, and the traversal hitting the traversal queue.